src/client/Client.cc

```diff
@@ -11114,6 +11114,11 @@ int64_t Client::_read(Fh *f, int64_t offset, uint64_t size, bufferlist *bl,
 {
   ceph_assert(ceph_mutex_is_locked_by_me(client_lock));
 
+  // zero bytes read is not supported by osd
+  if (size < 1) {
+    return -EINVAL;
```

Comment:
Can you provide more detail why an error code is returned versus a no-op? For example, if a read len is 0, I would expect the actual read to be skipped, an empty buffer to be returned, and a ret code of 0.

Reply:
`Striper::file_to_extents` asserts on anything <= 0 because of this treating this as failure rather than no-op.

Reply:
I think it would suffice to return 0 here rather than an err unless I'm missing something (discussion, maybe)?

Reply:
read fsync man page and it doesn't explicitly mention how the zero byte flush should be handled but by the looks of it does hint to a no-op. i'll make this to return `0` but now this makes me wonder if the _read call should return right away if size is zero or just handle 
```
if (f->flags & O_RSYNC) {
      _flush_range(in, offset, size);
}
```
but that would obv' leave the sync cal (case 3) exposed.